Libraries/Microsoft.Teams.Apps/Events/EventEmitter.cs

```diff
@@ -8,63 +8,75 @@ internal class EventEmitter
-        var topic = Topics.TryGetValue(name, out Topic? value) ? value : [];
+        if (!Topics.ContainsKey(name))
+        {
+            Topics[name] = [];
+        }
 
-        topic.Add(delegate (IPlugin plugin, Event @event, CancellationToken cancellationToken)
+        Topics[name].Add((plugin, @event, cancellationToken) =>
         {
             handler(plugin, @event);
             return Task.FromResult<object?>(null);
         });
 
-        Topics[name] = topic;
         return this;
     }
 
     public EventEmitter On<TResult>(string name, Func<IPlugin, Event, TResult> handler)
     {
-        var topic = Topics.TryGetValue(name, out Topic? value) ? value : [];
+        if (!Topics.ContainsKey(name))
+        {
+            Topics[name] = [];
+        }
 
-        topic.Add(delegate (IPlugin plugin, Event @event, CancellationToken cancellationToken)
+        Topics[name].Add((plugin, @event, cancellationToken) =>
         {
             var res = handler(plugin, @event);
             return Task.FromResult<object?>(res);
         });
 
-        Topics[name] = topic;
         return this;
     }
 
     public EventEmitter On(string name, Func<IPlugin, Event, CancellationToken, Task> handler)
     {
-        var topic = Topics.TryGetValue(name, out Topic? value) ? value : [];
+        if (!Topics.ContainsKey(name))
+        {
+            Topics[name] = [];
+        }
 
-        topic.Add(async delegate (IPlugin plugin, Event @event, CancellationToken cancellationToken)
+        Topics[name].Add(async (plugin, @event, cancellationToken) =>
         {
             await handler(plugin, @event, cancellationToken);
             return null;
         });
 
-        Topics[name] = topic;
         return this;
     }
 
     public EventEmitter On<TResult>(string name, Func<IPlugin, Event, CancellationToken, Task<TResult>> handler)
     {
-        var topic = Topics.TryGetValue(name, out Topic? value) ? value : [];
+        if (!Topics.ContainsKey(name))
+        {
+            Topics[name] = [];
+        }
 
-        topic.Add(async delegate (IPlugin plugin, Event @event, CancellationToken cancellationToken)
+        Topics[name].Add(async (plugin, @event, cancellationToken) =>
         {
             var res = await handler(plugin, @event, cancellationToken);
             return res;
         });
 
-        Topics[name] = topic;
         return this;
     }
 
-    public Task<object?> Emit(IPlugin plugin, string name, Event? @event = null, CancellationToken cancellationToken = default)
+    public async Task<object?> Emit(IPlugin plugin, string name, Event? @event = null, CancellationToken cancellationToken = default)
     {
-        var topic = Topics.TryGetValue(name, out Topic? value) ? value : [];
-        return topic.Emit(plugin, @event, cancellationToken);
+        if (!Topics.ContainsKey(name))
+        {
+            Topics[name] = [];
+        }
+
+        return await Topics[name].Emit(plugin, @event, cancellationToken);
```

Comment:
Really dumb question, but if Topics[name] is an empty array, how can we call emit on it? Do we have an extension on this empty array?

Reply:
Oh nvm i understand. It extends List, and has a function called emit got it that saves the handler